### ISSUE TITLE
Mark static fields generated by `[Instance]` as `readonly`

### ISSUE THREAD

**Issue by MatthijsMud** [SteveDunn/Vogen#315]
### Describe the feature

Vogen generates instances as mutable globals.

```C#
using System;
using Vogen;

namespace Example
{
  [ValueObject(typeof(Guid))]
  [Instance("Unspecified", "Guid.Empty")]
  public partial struct CustomerId {}
}
```

```C#
//…
public static CustomerId Unspecified = new CustomerId(Guid.Empty);
```

This allows a user to overwrite the generated values, which could result in unintended behavior.

```C#
var customers = new[] {/* … */};
CustomerId.Unspecified = CustomerId.From(Guid.NewGuid());
var customersWithoutId = customers.Where(customer => customer.Id == CustomerId.Unspecified).ToList();
```

The most common reason for overwriting the presets I can think of, is "by accident" (i.e. Yoda statement `if (CustomerId.Unspecified = Customer.Id)`). Having the values as readonly can cause the compiler to help catch those issues. There might however be some use cases I'm overlooking, which could be troublesome given this would be a breaking change.

My guess is that [InstanceGeneration.cs line 39](https://github.com/SteveDunn/Vogen/blob/06b0f7cdc1f99afd5bf0f444914b478be8826b53/src/Vogen/InstanceGeneration.cs#L39) is responsible.

**Comment by SteveDunn** [comment]
An excellent suggestion - thank you!

Yes, I agree that they should be readonly.  Thanks for the feedback. I'll implement this shortly.

I'm also looking at this whole mechanism of using attributes for this. I now think it was a mistake and I should've allowed users to specify their own values (_but only in the scope of the value object itself_), i.e. _allowing_ a Value Object to be `new`ed up inside the scope of the Value Object itself.

I have an open Issue for it [here](https://github.com/SteveDunn/Vogen/issues/221).

**Comment by SteveDunn** [comment]
@MatthijsMud - this should now be fixed in release 3.0.8 - thanks again for the feedback!
